packages/apps/desktop-viewer-test/src/frontend/components/AppComponent.tsx

```diff
@@ -0,0 +1,187 @@
+const makeLocalStoreName = (name: LocalStoreName) => `DesktopViewer:${name}`;
+const fromLocalStorage = (name: LocalStoreName) =>
+  window.localStorage.getItem(makeLocalStoreName(name)) ?? undefined; // replace null with undefined
+const toLocalStorage = (baseName: LocalStoreName, val?: string) => {
+  const name = makeLocalStoreName(baseName);
+  const storage = window.localStorage;
+  if (val === undefined || val.length === 0) {
+    storage.removeItem(name);
+  } else {
+    storage.setItem(name, val);
+  }
+};
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const AppComponent = () => {
+  const [projectId, setProjectId] = useState<string>();
+  const [iModelId, setIModelId] = useState<string>();
+  const [snapshotPath, setSnapshotPath] = useState<string>();
+  const [frontstages, setFrontstages] = useState<ViewerFrontstage[]>();
+  const [backstageItems, setBackstageItems] = useState<ViewerBackstageItem[]>();
+  const [isAuthorized, setIsAuthorized] = useState<boolean>(false);
+  const [connectedMode, setConnectedMode] = useState<boolean>(false);
+  const [frontstage, setFrontstage] =
+    useState<"IModelSelector" | "SnapshotSelector" | "Viewer">();
+
+  const selectedSnapshot = useSelector<RootState, string>(
+    (state: RootState) => state?.switchIModelState?.selectedSnapshot
+  );
+  const selectedIModel = useSelector<RootState, SelectedIModel | undefined>(
+    (state: RootState) => state?.switchIModelState?.selectedIModel
+  );
+  const switchSelectionState = useSelector<RootState, SwitchState>(
+    (state: RootState) => state?.switchIModelState?.switchState
+  );
+
+  const onIModelAppInitialized = async () => {
+    await IModelSelect.initialize(IModelApp.i18n);
+
+    const config = await ITwinViewerApp.getConfig();
+    if (config?.snapshotName) {
+      ITwinViewerApp.store.dispatch({
+        type: "App:OPEN_SNAPSHOT",
+        payload: config.snapshotName,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (connectedMode) {
+      IModelApp.authorizationClient?.onUserStateChanged.addListener(() => {
+        setIsAuthorized(IModelApp.authorizationClient?.isAuthorized || false);
+      });
+    }
+  }, [connectedMode]);
+
+  const openIModelSelector = async () => {
+    setConnectedMode(true);
+    setFrontstage("IModelSelector");
+    let frontstageDef;
+    if (isAuthorized) {
+      frontstageDef = FrontstageManager.findFrontstageDef("IModelSelector");
+    } else {
+      frontstageDef = FrontstageManager.findFrontstageDef("SignIn");
+    }
+    await FrontstageManager.setActiveFrontstageDef(frontstageDef);
+  };
+
+  const openSnapshotSelector = async () => {
+    setConnectedMode(false);
+    setFrontstage("SnapshotSelector");
+    const frontstageDef =
+      FrontstageManager.findFrontstageDef("SnapshotSelector");
+    await FrontstageManager.setActiveFrontstageDef(frontstageDef);
+  };
+
+  const initIModel = useCallback(() => {
+    if (selectedIModel) {
+      setFrontstage("Viewer");
```

Comment:
is it necessary to save these values to both state in the component and local storage?
ie instead of saved to app state, could just fetch the value from local storage with your fromLocalStorage fn

Reply:
I feel like app state and ls are mutually exclusive. Redux is used because these values are set in frontstages that are "disconnected" from the component tree for all intents and purposes. Ls is used to retain that last opened model for the next time that the app is opened